messaging/src/main/java/org/axonframework/deadline/jobrunnr/JobRunrDeadlineManager.java

```diff
@@ -0,0 +1,298 @@
+
+    private final ScopeAwareProvider scopeAwareProvider;
+    private final JobScheduler jobScheduler;
+    private final TransactionManager transactionManager;
+    private final SpanFactory spanFactory;
+
+    /**
+     * Instantiate a Builder to be able to create a {@link JobRunrDeadlineManager}.
+     * <p>
+     * The {@link TransactionManager} is defaulted to a {@link NoTransactionManager}.
+     * <p>
+     * The {@link SpanFactory} is defaulted to a {@link NoOpSpanFactory}.
+     * <p>
+     * The {@link JobScheduler} and {@link ScopeAwareProvider} are <b>hard requirements</b> and as such should be
+     * provided.
+     *
+     * @return a Builder to be able to create a {@link JobRunrDeadlineManager}
+     */
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    /**
+     * Instantiate a {@link JobRunrDeadlineManager} based on the fields contained in the
+     * {@link JobRunrDeadlineManager.Builder}.
+     * <p>
+     * Will assert that the {@link ScopeAwareProvider} and {@link JobScheduler} are not {@code null}, and will throw an
+     * {@link AxonConfigurationException} if any of them is {@code null}.
+     *
+     * @param builder the {@link QuartzDeadlineManager.Builder} used to instantiate a {@link QuartzDeadlineManager}
+     *                instance
+     */
+    protected JobRunrDeadlineManager(Builder builder) {
+        builder.validate();
+        this.scopeAwareProvider = builder.scopeAwareProvider;
+        this.jobScheduler = builder.jobScheduler;
+        this.transactionManager = builder.transactionManager;
+        this.spanFactory = builder.spanFactory;
+    }
+
+    @Override
+    public String schedule(@Nonnull Instant triggerDateTime, @Nonnull String deadlineName,
+                           @Nullable Object messageOrPayload,
+                           @Nonnull ScopeDescriptor deadlineScope) {
+        DeadlineMessage<Object> deadlineMessage = asDeadlineMessage(deadlineName, messageOrPayload, triggerDateTime);
+        UUID deadlineId = UUID.fromString(deadlineMessage.getIdentifier());
+        Span span = spanFactory.createDispatchSpan(() -> "JobRunrDeadlineManager.schedule(" + deadlineName + ")",
+                                                   deadlineMessage);
+        runOnPrepareCommitOrNow(span.wrapRunnable(() -> {
+            DeadlineMessage<Object> interceptedDeadlineMessage = processDispatchInterceptors(deadlineMessage);
+            DeadlineDetails deadlineDetails =
+                    new DeadlineDetails(deadlineName,
+                                        deadlineId,
+                                        deadlineScope,
+                                        interceptedDeadlineMessage.getPayload(),
+                                        new HashMap<>(interceptedDeadlineMessage.getMetaData()));
+            jobScheduler.<JobRunrDeadlineManager>schedule(deadlineId, triggerDateTime, x -> x.execute(deadlineDetails));
```

Comment:
Curious: So, the `JobScheduler` does all the serialization for us?

Reply:
Spotting some of the test cases, I assume this is configured on the `JobScheduler`, right? 
Does JobRunr provide a means so that we can define the serialized solution?

Although Jackson is a reasonable options, our users may have set up XStream as the serializer for everything. 
Having them configure another serializer for JobRunr specifically may be a bit "unfriendly."
What's your take on that, @gklijs?

Reply:
I'm not sure. it should be Json, so XStream is not going to work. I just tried Gson, by adding:
```
    @Bean
    public StorageProvider storageProvider(MongoClient mongoClient){
        MongoDBStorageProvider provider = new MongoDBStorageProvider(mongoClient);
        provider.setJobMapper(new JobMapper(new GsonJsonMapper()));
        return provider;
    }
```
But that ended up serialized with `Interfaces can\u0027t be instantiated! Register an InstanceCreator or a TypeAdapter for this type. Interface name: org.axonframework.messaging.ScopeDescriptor`. So we either can add to the documentation to use the jackson default, which works, or make the `Deadlinedetails` more serializer friendly, like it was in Tom's initial commit. We could also do both. I'm not sure how likely users would want to use something else. Using `JsonbJsonMapper` also fails because the ScopeDescriptor is directly used. I think we should use a configurable serializer so we could store the serialized value and the class, and use that to get the values back, similarly for the metadata. So those are just `String` fields, and whatever is choosen for jobrunner works?

Reply:
>  I think we should use a configurable serializer so we could store the serialized value and the class, and use that to get the values back, similarly for the metadata. So those are just String fields, and whatever is choosen for jobrunner works?

I like this suggestion. If we already serialize all the Axon-bits into `String` or perhaps `byte[]` for the deadline details object, JobRunr shouldn't have any issues with serializing it further into whatever serializer is set there.

Reply:
Turned out that was even a problem with jsonb. So now it also uses the configured serializer for the whole of the details, so just the bytes are used. Which means the serializer used with Jubrunr is not important anymore, as it always works now.

Reply:
Awesome news! Glad to have that cleared out.